corehq/apps/locations/templates/locations/manage/locations.html

```diff
@@ -226,3 +245,26 @@
 </div>
 {% endblock %}
 
+{% block modals %}
+{{ block.super }}
+<div class="modal hide fade" id="archive-location-modal">
+    <div class="modal-header">
+        <a class="close" data-dismiss="modal">&times;</a>
+        <h4>{% trans "Archive Location:" %} <span data-bind="text: name" \></h4>
+    </div>
+    <div class="modal-body">
+        <p>
+            {% blocktrans %}
+            <strong>Warning!</strong> Archiving a location will unassign any users
+            which were associated with that location.  You can unarchive this
+            location at any point, but you will have to reassign the users
+            manually.
+            {% endblocktrans %}
```

Comment:
I would change "unassign any users" to "remove location assignment for any users" and change "reassign the users" to "add the location back" or similar

cc @modonnell729